test/performance.ts

```diff
@@ -50,4 +50,67 @@ describe('performance', function() {
     const average = Math.floor((cycleCount * ipv6fixtures.length) / friendlyElapsed);
     console.log(`average IPv6 performance was ${average.toLocaleString()} per second`);
   });
+
+  // we keep this cache outside the tests, as it should be global
+  const checkerCache = new Map<string, ReturnType<typeof createChecker>>();
```

Comment:
This can be put in a `beforeEach` function. That has no noticeable effect on performance and keeps the tests isolated from each other.

```suggestion
    let checkerCache: Map<string, ReturnType<typeof createChecker>>;

    beforeEach(() => {
      checkerCache = new Map();
    });
```